### DISCUSSION TITLE
Question regarding the use of poetry in a CI workflow, and especially with caching

### DISCUSSION THREAD

**Discussion by Lucas-C** [python-poetry/poetry#4943] [Q&A]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

## Issue
I have setup a GitHub Actions workflow for a Pelican plugin, using `poetry` as a dependency manager:
https://github.com/pelican-plugins/linkbacks/blob/master/.github/workflows/continuous-integration-workflow.yml

I have a special constraint: I need to install `pelican` in editable mode, to gain access to its `tests.support` package. Because currently `poetry` does not support to install dependencies in editable mode, I install my `dev` dependencies through a `dev-requirements.txt` file.

My approach works, but is a bit clunky:
- in order to cache dependencies, because `poetry` stores its dependencies in a `virtualenv`, I need 2 caching steps for both `~/.cache/pip` & `~/.local/share/virtualenvs` directories
- in order to call `pylint`, `py.test` or install `dev-requirements.txt`, I need to either prefix all commands by `poetry run` or else call `source $(poetry env info --path)/bin/activate`

`poetry run` is fine, but in a multi-lines workflow-step, where several commands would need to be executed in the virtualenv, entering it once can quickly becomes more efficient and less verbose IMHO

The whole workflow feels verbose and a bit cumbersome.

Would you please have any suggestion to improve this using `poetry` ?

And a more specific questions: are `poetry` developpers considering a no-virtualenv option, to install dependencies in `~/.cache/pip`, which would make caching dependencies directories simpler in the case of a CI workflow, where there is not a need for virtual environments ?

**Comment by tony** [comment]
For GitHub actions users:

As of yesterday, `@actions/setup-python@v3` supports this via v3.1.0: [Release](https://github.com/actions/setup-python/releases/tag/v3.1.0), [Tag](https://github.com/actions/setup-python/tree/v3.1.0), https://github.com/actions/setup-python/pull/281

```yaml
steps:
- uses: actions/checkout@v3
- name: Install poetry
  run: pipx install poetry
- uses: actions/setup-python@v3
  with:
    python-version: '3.9'
    cache: 'poetry'
- run: poetry install
- run: poetry run pytest
```

If you see this via Google / Search and use gh actions, please search the tracker in case this ever changes.  See also: https://github.com/python-poetry/poetry/discussions/4205
